File: test/leetcode/TestPreorderTraversal.java

import org.junit.Test;
import static org.junit.Assert.*;
import org.junit.runner.JUnitCore;

import java.util.List;
import java.util.ArrayList;
import LeetCode.TreeNode;
import LeetCode.PreorderTraversal;

public class TestPreorderTraversal {
    private PreorderTraversal solution = new PreorderTraversal();

    @Test
    public void testTraverseEmptyTree() {
        ArrayList<Integer> expected = new ArrayList<Integer>();
        List<Integer> result = this.solution.preorderTraversal(null);
        assertEquals(result, expected);
    }

    @Test
    public void testTraverseOneElementTree() {
        ArrayList<Integer> expected = new ArrayList<Integer>();
        expected.add(1);
        List<Integer> result = this.solution.preorderTraversal(new TreeNode(1));
        assertEquals(result, expected);
    }

    @Test
    public void testTraverseTreeElementTree() {
        ArrayList<Integer> expected = new ArrayList<Integer>();
        expected.add(1);
        expected.add(2);
        expected.add(3);
        TreeNode root   = new TreeNode(1);
        root.right      = new TreeNode(2);
        root.right.left = new TreeNode(3);
        List<Integer> result = this.solution.preorderTraversal(root);
        assertEquals(result, expected);
    }

    @Test
    public void testTraverseFourElementTree() {
        ArrayList<Integer> expected = new ArrayList<Integer>();
        expected.add(1);
        expected.add(4);
        expected.add(2);
        expected.add(3);
        expected.add(5);
        TreeNode root   = new TreeNode(1);
        root.left       = new TreeNode(4);
        root.right      = new TreeNode(2);
        root.right.left = new TreeNode(3);
        root.right.right = new TreeNode(5);
        List<Integer> result = this.solution.preorderTraversal(root);
        assertEquals(result, expected);
    }
}

File: test/leetcode/test_preorder_traversal.py
#! /usr/bin/python
#

import init
import unittest

from preorder_traversal import TreeNode, Solution


class PreorderTraversalTestCase(unittest.TestCase):
    def setUp(self):
        self.solution = Solution()

    def testPreorderTraversalEmptyTree(self):
        result = self.solution.preorderTraversal(None)
        self.assertEqual(result, [])

    def testPreorderTraversal1(self):
        root = TreeNode(1)
        result = self.solution.preorderTraversal(root)
        self.assertEqual(result, [1])

    def testPreorderTraversal3Balanced(self):
        root = TreeNode(1)
        root.left = TreeNode(2)
        root.right = TreeNode(3)
        result = self.solution.preorderTraversal(root)
        self.assertEqual(result, [1,2,3])

    def testPreorderTraversal3Unbalanced(self):
        root = TreeNode(1)
        root.right = TreeNode(2)
        root.right.left = TreeNode(3)
        result = self.solution.preorderTraversal(root)
        self.assertEqual(result, [1,2,3])

    def testPreorderTraversal9(self):
        root = TreeNode(6)
        root.into([2, 1, 4, 3, 5, 7, 9, 8])
        result = self.solution.preorderTraversal(root)
        self.assertEqual(result, [6,2,1,4,3,5,7,9,8])


if __name__ == '__main__':
    unittest.main()
